include/nbl/builtin/hlsl/tgmath.hlsl

```diff
@@ -88,7 +88,7 @@ template<typename T>
 inline T floor(NBL_CONST_REF_ARG(T) val)
 {
 #ifdef __HLSL_VERSION
-    return spirv::floor(val);
+    return spirv::floor<T>(val);
 #else
     return glm::floor(val);
```

Comment:
aren't the GLM functions templated as well?

Reply:
@Przemog1 will take ownership

Reply:
The compiler doesn't seem to like using GLM templated for some reason.  
Gives a `no matching overloaded function found`.

Reply:
> The compiler doesn't seem to like using GLM templated for some reason. Gives a `no matching overloaded function found`.

what were you trying?

It might be because `floor` in `glm` has two overloads and is unfortunately templated differently for vectors
https://github.com/g-truc/glm/blob/33b4a621a697a305bc3a7610d290677b96beb181/glm/common.hpp#L65

Its probably better to use `std::floor` and forward per channel because GLM will refuse to work on any other vector type.

Reply:
leave as is, @Przemog1 will improve later.